File: src/main/java/it/unicam/cs/diciottoPolitico/casotto/model/interfaces/Ombrellone.java
package it.unicam.cs.diciottoPolitico.casotto.model.interfaces;

import it.unicam.cs.diciottoPolitico.casotto.model.Categoria;
import it.unicam.cs.diciottoPolitico.casotto.utils.QRCode;

import java.util.UUID;

/**
 * Rappresenta un ombrellone.
 */
public interface Ombrellone {

    /**
     * Ritorna l'id dell'ombrellone.
     *
     * @return id dell'ombrellone
     */
    UUID getId();

    /**
     * Ottieni il codice identificativo dell'ombrellone sulla spieggia
     *
     * @return il codice dell'ombrellone
     */
    QRCode getCodiceSpiaggia();

    /**
     * Imposta il codice della spiaggia.
     *
     * @param codiceSpiaggia il nuovo codice della spiaggia
     */
    void setCodiceSpiaggia(QRCode codiceSpiaggia);

    /**
     * Ritorna la {@link Categoria} dell'ombrellone.
     *
     * @return la categoria dell'ombrellone
     */
    Categoria getCategoria();

    /**
     * Imposta la categoria dell'ombrellone.
     *
     * @param categoria la nuova categoria dell'ombrellone
     * @throws NullPointerException se la categoria &egrave; {@code null}
     */
    void setCategoria(Categoria categoria);
}

File: Casotto-Front-End/src/app/ombrellone.ts
import { CategoriaOmbrellone } from "./categoria-ombrellone";

/*export interface Ombrellone{
    //id: string,
    catagoria: CategoriaOmbrellone,
    codiceSpiaggia: string
}*/

export class Ombrellone{
    
    constructor(private _categoria: CategoriaOmbrellone,
                private _codiceSpiaggia: string){}

    //get id(): number { return this._id }
    //set id(id: number) { this._id = id}

    get categoria(): CategoriaOmbrellone { return this._categoria }
    set categoria(catagoria: CategoriaOmbrellone) { this._categoria = catagoria }

    get codiceSpiaggia(): string { return this._codiceSpiaggia }
    set codiceSpiaggia(codiceSpiaggia: string) { this._codiceSpiaggia = codiceSpiaggia }
}